### PR TITLE
Add vcpkg cache secrets for C++

### PR DESCRIPTION
**Pull request by danieljurek**

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
<3

File: eng/pipelines/pipeline-generation.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ parameters:
     default: 76
   - name: APIReview_AutoCreate_Configurations
     default: 93
+  - name: Cache_Secrets_for_CPP
```
Comment (weshaggard) [inline eng/pipelines/pipeline-generation.yml:29]: What is the secret for? Are we not using DevOps caching any longer?

Comment (danieljurek) [reply eng/pipelines/pipeline-generation.yml:29]: The intent is to move away from Azure DevOps caching -- https://github.com/Azure/azure-sdk-for-cpp/pull/3329

**Comment by danieljurek** [comment]
This file also needs to change to include the variable group -- https://github.com/Azure/azure-sdk-for-cpp/blob/main/eng/pipelines/prepare-pipelines.yml#L9

**Comment by danieljurek** [comment]
Don't use the Test KV, use the EngSys KV.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**